server/src/lib/server.rs

```diff
@@ -315,6 +324,24 @@ async fn instance_ensure(
                 info!(rqctx.log, "Disk {} created successfully", disk.name);
             }
 
+            if let Some(cloud_init_bytes) = &cloud_init_bytes {
+                info!(rqctx.log, "Creating cloud-init disk");
+                let bdf = slot_to_bdf(api::Slot(0), SlotType::CloudInit)
+                    .map_err(|e| {
+                        Error::new(ErrorKind::InvalidData, e.to_string())
+                    })?;
+
+                let bytes = base64::decode(&cloud_init_bytes).map_err(|e| {
+                    Error::new(ErrorKind::InvalidInput, e.to_string())
+                })?;
+
+                init.initialize_in_memory_virtio_from_bytes(
+                    &chipset, &bytes, bdf, true,
```

Comment:
Given this is the only use of `initialize_in_memory_virtio_from_bytes` should we just pass the vec to avoid the subsequent `to_vec` call

Reply:
:+1: fixed in 4b883c2